From 059626cd9c525ac715ab73cce2d9b3e2ecc7ce53 Mon Sep 17 00:00:00 2001
From: Mattias Busck <mattias.busck@hostmobility.com>
Date: Tue, 30 May 2023 15:11:34 +0200
Subject: [PATCH 4/4] tcan4x4 stop queue first in transmit to avoid race

---
 drivers/net/can/m_can/m_can.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/net/can/m_can/m_can.c b/drivers/net/can/m_can/m_can.c
index d3f29ab2142a..c20b64906352 100644
--- a/drivers/net/can/m_can/m_can.c
+++ b/drivers/net/can/m_can/m_can.c
@@ -1680,6 +1680,7 @@ static netdev_tx_t m_can_tx_handler(struct m_can_classdev *cdev)
 	} else {
 		/* Transmit routine for version >= v3.1.x */
 
+		netif_stop_queue(dev);
 		txfqs = m_can_read(cdev, M_CAN_TXFQS);
 
 		/* Check if FIFO full */
@@ -1735,9 +1736,8 @@ static netdev_tx_t m_can_tx_handler(struct m_can_classdev *cdev)
 		m_can_write(cdev, M_CAN_TXBAR, (1 << putidx));
 
 		/* stop network queue if fifo full */
-		if (m_can_tx_fifo_full(cdev) ||
-		    m_can_next_echo_skb_occupied(dev, putidx))
-			netif_stop_queue(dev);
+		//if (m_can_tx_fifo_full(cdev) ||
+		    //m_can_next_echo_skb_occupied(dev, putidx))
 	}
 
 	return NETDEV_TX_OK;
-- 
2.30.2

