diff --git a/arch/arm64/boot/dts/freescale/imx8mp-var-dart-hmx1.dts b/arch/arm64/boot/dts/freescale/imx8mp-var-dart-hmx1.dts
index ed20a2c76d8f..bf3cb7531aa2 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp-var-dart-hmx1.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mp-var-dart-hmx1.dts
@@ -9,12 +9,21 @@
 
 / {
 	model = "Host Mobility HMX1 imx8mp";
+	comment = "eth1 fixed back wednesday 28 dec maybe not disable. phy 3/7 no reset";
 
 	aliases {
 		rtc0 = &rtc_extern;
 		rtc1 = &snvs_rtc;
 		ethernet2 = &obdnet;
 		ethernet3 = &usbnet;
+		/*
+		can0 = &can0;
+		can1 = &can1;
+		can2 = &can2;
+		can3 = &can3;
+		can4 = &can4;
+		can5 = &can5;
+		*/
 	};
 
 	chosen {
@@ -24,6 +33,7 @@ chosen {
 	};
 
 	reg_usb_hubs: regulator-usbhubs {
+		comment = "USB hubs high";
 		compatible = "regulator-fixed";
 		regulator-name = "regulator-usb-hubs";
 		gpio = <&gpio1 12 GPIO_ACTIVE_HIGH>;
@@ -32,9 +42,9 @@ reg_usb_hubs: regulator-usbhubs {
 		pinctrl-0 = <&pinctrl_usbhubs>;
 		pinctrl-names = "default";
 
-		regulator-state-mem {
-			regulator-on-in-suspend;
-		};
+		//regulator-state-mem {
+	//		regulator-on-in-suspend;
+		//};
 	};
 
 	gpio-keys {
@@ -248,16 +258,24 @@ &fec {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_fec>;
 	phy-mode = "rgmii";
-	phy-handle = <&ethphy_t1_1>;
+	//phy-handle = <&ethphy_t1_1>;
 	status = "okay";
+	//phy-reset-gpios = <&gpio3 1 GPIO_ACTIVE_HIGH>;
+	fixed-link {
+		speed = <100>;
+		full-duplex;
+	};
 	mdio {
 		#address-cells = <1>;
 		#size-cells = <0>;
+		/* try with fixed link
 		ethphy_t1_1: ethernet-phy@0 {
 			// trying med clause 22 instead of clause 45
+			// qqq: check what happens now compatible = "ethernet-phy-ieee802.3-c45";
+			//compatible = "marvell";
 			compatible = "ethernet-phy-ieee802.3-c45";
 			reg = <3>;
-			reset-gpios = <&gpio3 1 GPIO_ACTIVE_HIGH>;
+			//reset-gpios = <&gpio3 1 GPIO_ACTIVE_HIGH>;
 			at803x,eee-disabled;
 			eee-broken-1000t;
 			reset-assert-us = <10000>;
@@ -269,18 +287,26 @@ vddio1: vddio-regulator {
 				regulator-max-microvolt = <1800000>;
 			};
 		};
+		*/
+		ethphy_t1_2: ethernet-phy@1 {
+			// trying med clause 22 instead of clause 45
+			compatible = "ethernet-phy-ieee802.3-c45";
+			reg = <7>;
+		};
 	};
 };
 
 /* Second Automotive Ethernet Marvell 88Q2110*/
 &eqos {
-	mdio {
-		ethernet-phy@0 {
-			// trying med clause 22 instead of clause 45
-			compatible = "ethernet-phy-ieee802.3-c45";
-			reg = <3>;
-		};
+	phy-handle = <&ethphy_t1_2>;
+	fixed-link {
+		speed = <100>;
+		full-duplex;
+	};
+	/* MAYBE NOT DISABLE mdio {
+		status = "disabled";
 	};
+	*/
 };
 
 &i2c2 {
@@ -439,7 +465,8 @@ &usb3_0 {
 };
 
 &usb_dwc3_0 {
-	dr_mode = "otg";
+	comment = "testing with host on usb_dwc3_0";
+	dr_mode = "host";
 	hnp-disable;
 	srp-disable;
 	adp-disable;
@@ -464,8 +491,9 @@ &usb3_phy1 {
 
 &usb3_1 {
 	status = "okay";
+	comment = "no compatible";
 	lan9513i@1 {
-		compatible = "usb424,9514";
+		// selfchoice compatible = "usb424,9514";
 		reg =<1>;
 		#address-cells = <1>;
 		#size-cells = <0>;
@@ -473,7 +501,7 @@ lan9513i@1 {
 		vdd-supply = <&reg_usb_hubs>;
 
 		usbnet: usbether@1 {
-			compatible = "usb424,ec00";
+			// selfchoice compatible = "usb424,ec00";
 			reg =<1>;
 			local-mac-address = [ 00 00 00 00 00 00 ];
 			wakeup-source;
@@ -481,14 +509,14 @@ usbnet: usbether@1 {
 
 		// Cascaded LAN9513 (on USB3_D_P in schematics)
 		lan9513i@2 {
-			compatible = "usb424,9514";
+			// selfchoice compatible = "usb424,9514";
 			reg =<2>;
 			#address-cells = <1>;
 			#size-cells = <0>;
 			wakeup-source;
 
 			obdnet: obdether@1 {
-				compatible = "usb424,ec00";
+				// selfchoice compatible = "usb424,ec00";
 				reg =<1>;
 				local-mac-address = [ 00 00 00 00 00 00 ];
 				wakeup-source;
